manojgupta: Upstream patch modified to apply to eigen 3.3.7,
https://crbug.com/1059030.

From f0ce88cff75e2020b4beae54b094c77c526724a2 Mon Sep 17 00:00:00 2001
From: Tobias Bosch <tbosch@google.com>
Date: Mon, 24 Feb 2020 23:09:36 +0000
Subject: [PATCH] Include <sstream> explicitly, and don't rely on the implicit
 include via <complex>.

This implicit dependency does no longer exist in a recent llbm release (sha 78be61871704).
---
 Eigen/Core                         | 1 +
 Eigen/src/Core/arch/Default/Half.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/Eigen/Core b/Eigen/Core
index b923b8c..39d572a 100644
--- a/Eigen/Core
+++ b/Eigen/Core
@@ -279,6 +279,7 @@
 #include <cmath>
 #include <cassert>
 #include <functional>
+#include <sstream>
 #include <iosfwd>
 #include <cstring>
 #include <string>
diff --git a/Eigen/src/Core/arch/CUDA/Half.h b/Eigen/src/Core/arch/CUDA/Half.h
index 755e620..10318dd 100644
--- a/Eigen/src/Core/arch/CUDA/Half.h
+++ b/Eigen/src/Core/arch/CUDA/Half.h
@@ -42,6 +42,7 @@
 #define EIGEN_EXPLICIT_CAST(tgt_type) operator tgt_type()
 #endif
 
+#include <iostream>
 
 namespace Eigen {
 
